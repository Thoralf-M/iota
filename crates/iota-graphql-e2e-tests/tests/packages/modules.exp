processed 6 tasks

task 1, lines 7-32:
//# publish
created: object(1,0)
mutated: object(0,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 6011600,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, line 34:
//# create-checkpoint
Checkpoint created: 1

task 3, lines 36-66:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x9c1d6aa32c63a2dcb3bab3eacc66a926310b03a0e524bd5e3332cfc71fc13150",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd8f043ea3e826fc23937bcb55eeafcc746bb666a70104af624fbf8a382a8da8f",
                    "asMovePackage": {
                      "module": {
                        "name": "m",
                        "package": {
                          "address": "0xd8f043ea3e826fc23937bcb55eeafcc746bb666a70104af624fbf8a382a8da8f"
                        },
                        "fileFormatVersion": 6,
                        "bytes": "oRzrCwYAAAAIAQAGAgYKAxARBCEEBSUfB0QkCGhADKgBMAAGAQMBBQEADAEAAQIBAgAABAABAQIAAgIBAAEHBQEBAAIEAAYCAwYLAAEJAAEDAQYLAAEIAQABCQABBgsAAQkAAQgBBENvaW4ESU9UQQNiYXIEY29pbgNmb28EaW90YQFtBXZhbHVl2PBD6j6Cb8I5N7y1Xur8x0a7ZmpwEEr2JPv4o4Ko2o8AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgABAAADBQsBOAALABYCAQEAAAMIBioAAAAAAAAACgA4AQYrAAAAAAAAAAsAOAEYAgA=",
                        "disassembly": "// Move bytecode v6\nmodule d8f043ea3e826fc23937bcb55eeafcc746bb666a70104af624fbf8a382a8da8f.m {\nuse 0000000000000000000000000000000000000000000000000000000000000002::coin;\nuse 0000000000000000000000000000000000000000000000000000000000000002::iota;\n\n\n\n\n\n\npublic foo<Ty0: drop>(Arg0: u64, Arg1: &Coin<Ty0>): u64 {\nB0:\n\t0: MoveLoc[1](Arg1: &Coin<Ty0>)\n\t1: Call coin::value<Ty0>(&Coin<Ty0>): u64\n\t2: MoveLoc[0](Arg0: u64)\n\t3: Add\n\t4: Ret\n\n}\npublic bar(Arg0: &Coin<IOTA>): u64 {\nB0:\n\t0: LdU64(42)\n\t1: CopyLoc[0](Arg0: &Coin<IOTA>)\n\t2: Call foo<IOTA>(u64, &Coin<IOTA>): u64\n\t3: LdU64(43)\n\t4: MoveLoc[0](Arg0: &Coin<IOTA>)\n\t5: Call foo<IOTA>(u64, &Coin<IOTA>): u64\n\t6: Mul\n\t7: Ret\n\n}\n}"
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 4, lines 68-105:
//# run-graphql --cursors {"n":"m","c":1} {"n":"o","c":1}
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x9c1d6aa32c63a2dcb3bab3eacc66a926310b03a0e524bd5e3332cfc71fc13150",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd8f043ea3e826fc23937bcb55eeafcc746bb666a70104af624fbf8a382a8da8f",
                    "asMovePackage": {
                      "all": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": false
                        }
                      },
                      "after": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": true
                        }
                      },
                      "before": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": false
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 5, lines 107-147:
//# run-graphql --cursors {"n":"m","c":1} {"n":"o","c":1}
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x9c1d6aa32c63a2dcb3bab3eacc66a926310b03a0e524bd5e3332cfc71fc13150",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd8f043ea3e826fc23937bcb55eeafcc746bb666a70104af624fbf8a382a8da8f",
                    "asMovePackage": {
                      "prefix": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": true
                        }
                      },
                      "prefixAll": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": true
                        }
                      },
                      "prefixExcess": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": true
                        }
                      },
                      "suffix": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": true
                        }
                      },
                      "suffixAll": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": false
                        }
                      },
                      "suffixExcess": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": false
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
