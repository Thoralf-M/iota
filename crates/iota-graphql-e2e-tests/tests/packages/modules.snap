---
source: external-crates/move/crates/move-transactional-test-runner/src/framework.rs
---
processed 6 tasks

task 1, lines 7-32:
//# publish
created: object(1,0)
mutated: object(0,0)
gas summary: computation_cost: 1000000, computation_cost_burned: 1000000, storage_cost: 6011600,  storage_rebate: 0, non_refundable_storage_fee: 0

task 2, line 34:
//# create-checkpoint
Checkpoint created: 1

task 3, lines 36-66:
//# run-graphql
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x6d72312db992499a342d70e04c177cfc9cb0dc264b0792c9d7bc10e0cc584940",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd6a6f9571d194335b08e9d94cb89c1cd3a9c383caf1bf74e399076657e0fdba4",
                    "asMovePackage": {
                      "module": {
                        "name": "m",
                        "package": {
                          "address": "0xd6a6f9571d194335b08e9d94cb89c1cd3a9c383caf1bf74e399076657e0fdba4"
                        },
                        "fileFormatVersion": 6,
                        "bytes": "oRzrCwYAAAAIAQAGAgYKAxARBCEEBSUfB0QkCGhADKgBMAAGAQMBBQEADAEAAQIBAgAABAABAQIAAgIBAAEHBQEBAAIEAAYCAwYLAAEJAAEDAQYLAAEIAQABCQABBgsAAQkAAQgBBENvaW4ESU9UQQNiYXIEY29pbgNmb28EaW90YQFtBXZhbHVl1qb5Vx0ZQzWwjp2Uy4nBzTqcODyvG/dOOZB2ZX4P26QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgABAAADBQsBOAALABYCAQEAAAMIBioAAAAAAAAACgA4AQYrAAAAAAAAAAsAOAEYAgA=",
                        "disassembly": "// Move bytecode v6\nmodule d6a6f9571d194335b08e9d94cb89c1cd3a9c383caf1bf74e399076657e0fdba4.m {\nuse 0000000000000000000000000000000000000000000000000000000000000002::coin;\nuse 0000000000000000000000000000000000000000000000000000000000000002::iota;\n\npublic foo<Ty0: drop>(Arg0: u64, Arg1: &Coin<Ty0>): u64 {\nB0:\n\t0: MoveLoc[1](Arg1: &Coin<Ty0>)\n\t1: Call coin::value<Ty0>(&Coin<Ty0>): u64\n\t2: MoveLoc[0](Arg0: u64)\n\t3: Add\n\t4: Ret\n}\n\npublic bar(Arg0: &Coin<IOTA>): u64 {\nB0:\n\t0: LdU64(42)\n\t1: CopyLoc[0](Arg0: &Coin<IOTA>)\n\t2: Call foo<IOTA>(u64, &Coin<IOTA>): u64\n\t3: LdU64(43)\n\t4: MoveLoc[0](Arg0: &Coin<IOTA>)\n\t5: Call foo<IOTA>(u64, &Coin<IOTA>): u64\n\t6: Mul\n\t7: Ret\n}\n\n}\n"
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 4, lines 68-105:
//# run-graphql --cursors {"n":"m","c":1} {"n":"o","c":1}
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x6d72312db992499a342d70e04c177cfc9cb0dc264b0792c9d7bc10e0cc584940",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd6a6f9571d194335b08e9d94cb89c1cd3a9c383caf1bf74e399076657e0fdba4",
                    "asMovePackage": {
                      "all": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": false
                        }
                      },
                      "after": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": true
                        }
                      },
                      "before": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": false
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

task 5, lines 107-147:
//# run-graphql --cursors {"n":"m","c":1} {"n":"o","c":1}
Response: {
  "data": {
    "transactionBlocks": {
      "nodes": [
        {
          "effects": {
            "objectChanges": {
              "nodes": [
                {
                  "outputState": {
                    "address": "0x6d72312db992499a342d70e04c177cfc9cb0dc264b0792c9d7bc10e0cc584940",
                    "asMovePackage": null
                  }
                },
                {
                  "outputState": {
                    "address": "0xd6a6f9571d194335b08e9d94cb89c1cd3a9c383caf1bf74e399076657e0fdba4",
                    "asMovePackage": {
                      "prefix": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": true
                        }
                      },
                      "prefixAll": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": true
                        }
                      },
                      "prefixExcess": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibyIsImMiOjF9",
                            "node": {
                              "name": "o"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": false,
                          "hasPreviousPage": true
                        }
                      },
                      "suffix": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": true
                        }
                      },
                      "suffixAll": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": false
                        }
                      },
                      "suffixExcess": {
                        "edges": [
                          {
                            "cursor": "eyJuIjoibSIsImMiOjF9",
                            "node": {
                              "name": "m"
                            }
                          },
                          {
                            "cursor": "eyJuIjoibiIsImMiOjF9",
                            "node": {
                              "name": "n"
                            }
                          }
                        ],
                        "pageInfo": {
                          "hasNextPage": true,
                          "hasPreviousPage": false
                        }
                      }
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
